core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2904,6 +2908,12 @@ private void registerCreateTableQuery(SqlValidatorScope parentScope,
     } else {
       throw newValidationError(createTable, RESOURCE.createTableRequiresAsQuery());
     }
+    // Store the scope of the create table node itself.
+    // This will be used later during validation to resolve table aliases and/or
+    // the associated sub query.
+    // Since the create table node is always the topmost node, I believe that the usingScope
```

Comment:
Can we confirm this? Let's not have comments that say "I believe"

Reply:
I can now confirm this, updated the comment.